metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/SurroundingStrategy.java

```diff
@@ -0,0 +1,152 @@
+import org.sentrysoftware.metricshub.engine.extension.ExtensionManager;
+import org.sentrysoftware.metricshub.engine.strategy.source.OrderedSources;
+import org.sentrysoftware.metricshub.engine.telemetry.TelemetryManager;
+
+/**
+ * Strategy for executing tasks that surround the core processing of sources, such as "beforeAll"
+ * and "afterAll" operations. Inherits from {@link AbstractStrategy} and manages source execution
+ * for a given {@link Connector}.
+ * <p>
+ * Leverages Lombok's {@link Slf4j} for logging, {@link Data} for auto-generating getters and setters,
+ * and {@link EqualsAndHashCode} to ensure proper equality and hash code generation.
+ */
+@Slf4j
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class SurroundingStrategy extends AbstractStrategy {
+
+	private Connector connector;
+
+	/**
+	 * Constructs a {@code SurroundingStrategy} with the necessary components for handling
+	 * "beforeAll" and "afterAll" source processing.
+	 *
+	 * @param telemetryManager  The manager responsible for telemetry (monitors and metrics).
+	 * @param strategyTime      The time allocated for strategy execution.
+	 * @param clientsExecutor   Executor service for managing client-related tasks.
+	 * @param connector         The connector defining sources and their dependencies.
+	 * @param extensionManager  The manager handling various extensions for strategy execution.
+	 */
+	@Builder
+	public SurroundingStrategy(
+		@NonNull final TelemetryManager telemetryManager,
+		@NonNull final Long strategyTime,
+		@NonNull final ClientsExecutor clientsExecutor,
+		@NonNull final Connector connector,
+		@NonNull final ExtensionManager extensionManager
+	) {
+		super(telemetryManager, strategyTime, clientsExecutor, extensionManager);
+		this.connector = connector;
+	}
+
+	/**
+	 * Executes the "beforeAll" and "afterAll" source processing for the configured connector.
+	 * This method logs relevant information and processes sources in a specific order based on their dependencies.
+	 * <p>
+	 * If no sources are available for processing, the method logs this information and returns early.
+	 */
+	@Override
+	public void run() {
+		// Retrieve the connector's identifier and hostname for logging and processing.
+		final String connectorId = connector.getCompiledFilename();
+		final String hostname = telemetryManager.getHostname();
+
+		// Fetch beforeAll sources and afterAll sources from the connector.
+		final Map<String, Source> beforeAllSources = connector.getBeforeAll();
+		final Map<String, Source> afterAllSources = connector.getAfterAll();
+
+		if (
+			(beforeAllSources == null || beforeAllSources.isEmpty()) && (afterAllSources == null || afterAllSources.isEmpty())
+		) {
+			log.debug(
+				"Hostname {} - Attempted to process beforeAll and afterAll sources, but none are available for connector {}.",
+				hostname,
+				connectorId
+			);
+			return;
+		}
+
+		if (beforeAllSources == null || beforeAllSources.isEmpty()) {

```

Comment:
🤦‍♂️We have already tested `(beforeAllSources == null || beforeAllSources.isEmpty())`